api/common/crypto.cc

```diff
@@ -0,0 +1,266 @@
+/* Copyright 2017 Outscale SAS
+ *
+ * This file is part of Butterfly.
+ *
+ * Butterfly is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published
+ * by the Free Software Foundation.
+ *
+ * Butterfly is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butterfly.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <cstring>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <memory>
+#include "api/common/crypto.h"
+
+namespace Crypto {
+
+#define AES_IV_SIZE 16
+#define AES_BLOCK_SIZE 16
+#define AES256_KEY_SIZE 32
+#define SHA512_SIZE 64
+
+bool LoadKeyFromPath(const std::string &path, std::string *key) {
+    std::ifstream file(path);
+    if (!file)
+        return false;
+    file.seekg(0, std::ios::end);
+    std::streampos length = file.tellg();
+    if (length != AES256_KEY_SIZE)
+        return false;
+    file.seekg(0, std::ios::beg);
+    char b[AES256_KEY_SIZE];
+    file.read(b, AES256_KEY_SIZE);
+    key->reserve(AES256_KEY_SIZE);
```

Comment:
As std::string implement the operator=( const Char* s );
I'm pretty sure, 
```key = b;```
can replace  
```
key->reserve(AES256_KEY_SIZE);
for (int i = 0; i < AES256_KEY_SIZE; i++)
    *key += b[i];
```